### ISSUE TITLE
Need to review failures around package install/update/delete tests

### ISSUE THREAD

**Issue by sghai** [SatelliteQE/robottelo#3642]
Following test are failing and looks like all failing while installing package on remote host

```
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_install_package  1 min 47 sec    2
 tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_install_package_group   1 min 48 sec    2
 tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_upgrade_package 1 min 51 sec    2
 tests.foreman.ui.test_hostcollection.HostCollectionPackageManagementTest.test_positive_install_package 12 min  2
 tests.foreman.ui.test_hostcollection.HostCollectionPackageManagementTest.test_positive_install_package_group   8 min 32 sec    2
 tests.foreman.ui.test_hostcollection.HostCollectionPackageManagementTest.test_positive_remove_package_group    8 min 27 sec    2
 tests.foreman.ui.test_hostcollection.HostCollectionPackageManagementTest.test_positive_upgrade_package 8 min 27 sec    2
```

**Comment by renzon** [comment]
I tried to tackle this one but am stuck when [creating vm](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_host.py#L944). It seem private key from provisioning server is missing in my environment, causing error when [getting connection](https://github.com/SatelliteQE/robottelo/blob/master/robottelo/ssh.py#L232). Maybe I should configure some private key to be able to access provisioning_server through ssh:

self = <paramiko.client.SSHClient object at 0x7fa1b1c43c10>, username = 'root'
password = None, pkey = None, key_filenames = ['/home/renzo/.ssh/id_rsa']
allow_agent = True, look_for_keys = True, gss_auth = False, gss_kex = False
gss_deleg_creds = True, gss_host = '<provisioning_server>'

>   ???
E   PasswordRequiredException: Private key file is encrypted

**Comment by renzon** [comment]
@abalakh helped me with permissions. So know I got another error:

```console
if options['subscription'] not in (sub['name'] for sub in subscriptions):
            raise CLIFactoryError(
                u'Subscription {0} not found in the given org'
>               .format(options['subscription'])
            )
E           CLIFactoryError: Subscription Red Hat Enterprise Linux Server Entry Level, Self-support not found in the given org

PycharmProjects/robottelo/robottelo/cli/factory.py:1759: CLIFactoryError
```

Debugging I see subscription is not available:
```python
options['subscription']='Red Hat Enterprise Linux Server Entry Level, Self-support'
```
and 

```python
[sub['name'] for sub in subscriptions]=<type 'list'>: [
u'Red Hat Enterprise Linux for Virtual Datacenters, Premium', 
u'Red Hat Enterprise Linux Server, Premium (Physical or Virtual Nodes)', 
u'Red Hat Enterprise Linux Atomic Host', 
u'Red Hat Enterprise Linux Server, Premium (8 sockets) (Unlimited guests)'
]
```

So a different subscription is need. What should I do @sghai ?

**Comment by omaciel** [comment]
@sghai to work with @renzon on this

**Comment by renzon** [comment]
@sghai I solved the subscription issue. Now I am getting error on setupClass while enabling repository:

```console
hammer> repository-set enable --product="Red Hat Enterprise Linux Server" --basearch="x86_64" --name="Red Hat Satellite Tools 6.2 (for RHEL 7 Server) (RPMs)" --organization-id="9"
Could not enable repository:
  Error: product not found
```
Any clue?

**Comment by renzon** [comment]
Ok, finally I was able to reproduce the error for tests on KatelloAgentTestCase. On jenkins test was failing during setUpClass because promoting ContentView was not possible. This bug was present on sat 6.2.5 but fixed on 6.2.6 (thanks @JacobCallahan for the info). So I upgraded my box and tests passed on my local version of robottelo:

```console
/home/renzo/PycharmProjects/venvs/robottelo/bin/python /home/renzo/bin/pycharm-2016.3/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/renzo/PycharmProjects/robottelo/tests/foreman/cli/test_host.py "-k KatelloAgentTestCase"
Testing started at 5:17 PM ...
============================= test session starts ==============================

================== 9 passed, 37 deselected in 1816.30 seconds ==================
```

**Comment by renzon** [comment]
On HostCollectionPackage install_package is working but install_package_group dont:

```console
/home/renzo/PycharmProjects/venvs/robottelo/bin/python /home/renzo/bin/pycharm-2016.3/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/renzo/PycharmProjects/robottelo/tests/foreman/ui/test_hostcollection.py "-k HostCollectionPackageManagementTest and test_positive_install_package"
============================= test session starts ==============================
    @tier3
    def test_positive_install_package_group(self):
        """Install a package group to hosts inside host collection remotely
    
            @id: 2bf47798-d30d-451a-8de5-bc03bd8b9a48
    
            @Assert: Package group was successfully installed on all the hosts in
            host collection
    
            @CaseLevel: System
            """
        with Session(self.browser):
            self.hostcollection.execute_bulk_package_action(
                self.host_collection.name,
                'install',
                'package_group',
                FAKE_0_CUSTOM_PACKAGE_GROUP_NAME,
            )
            for package in FAKE_0_CUSTOM_PACKAGE_GROUP:
>               self._validate_package_installed(self.hosts, package)

test_hostcollection.py:600: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test_hostcollection.py:505: in _validate_package_installed
    host.hostname,
E   AssertionError: Package cockateel-3.1-1.noarch was not installed on  host 139911699363472.lab.eng.rdu2.redhat.com
============= 1 failed, 1 passed, 20 deselected in 1228.89 seconds =============
```

I am going to investigate failing test after other 2 tests

**Comment by renzon** [comment]
remove_package_group failing on same part as install:

```console
        with Session(self.browser):
            self.hostcollection.execute_bulk_package_action(
                self.host_collection.name,
                'install',
                'package_group',
                FAKE_0_CUSTOM_PACKAGE_GROUP_NAME,
            )
            for package in FAKE_0_CUSTOM_PACKAGE_GROUP:
>               self._validate_package_installed(self.hosts, package)

test_hostcollection.py:621: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test_hostcollection.py:505: in _validate_package_installed
    host.hostname,
E   AssertionError: Package cockateel-3.1-1.noarch was not installed on  host 140046984549712.lab.eng.rdu2.redhat.com
============================= 21 tests deselected ==============================
================== 1 failed, 21 deselected in 923.04 seconds ===================
```

**Comment by renzon** [comment]
upgrade_package worked:

```console
/home/renzo/PycharmProjects/venvs/robottelo/bin/python /home/renzo/bin/pycharm-2016.3/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/renzo/PycharmProjects/robottelo/tests/foreman/ui/test_hostcollection.py "-k HostCollectionPackageManagementTest and test_positive_upgrade_package"
Testing started at 9:34 AM ...
============================= test session starts ==============================
================== 1 passed, 21 deselected in 676.43 seconds ===================
```

**Comment by renzon** [comment]
both failing tests have the same reason: installing group is not working. Investigating, this is the group been installed: https://github.com/SatelliteQE/robottelo/blob/7db1851110afda569c01e8db381c69eb0c3494bf/robottelo/constants.py#L492-L492. I have tried manually installing it and system show success msg. But it say installation is scheduled, but nothing is present on Task Monitor and packages (https://github.com/SatelliteQE/robottelo/blob/7db1851110afda569c01e8db381c69eb0c3494bf/robottelo/constants.py#L503) were not installed

**Comment by sghai** [comment]
@renzo: monitor /var/log/messages on the system where you'r installing the packages to see if any error/exception appears.

**Comment by renzon** [comment]
@sghai this is the log content:

```console
Jan 17 08:19:30 cloud-qe-12 pulp: pulp.server.agent.direct.services:INFO: (22600-37696) {'reboot': {'scheduled': False, 'details': {}}, 'details': {'package_group': {'details': {'message': 'No Group named birds exists', 'trace': 'Traceback (most recent call last):\n\n  File "/usr/lib/python2.7/site-packages/pulp/agent/lib/dispatcher.py", line 61, in in
```

**Comment by sghai** [comment]
@renzo: as we discussed y'day, the repo being used by the tests doesn't include any package-groups, so tests are throwing `No Group named birds exists`.

Tests are using this [FAKE_6_REPO]( https://github.com/SatelliteQE/robottelo/blob/7db1851110afda569c01e8db381c69eb0c3494bf/robottelo/constants.py#L464): (it may be good for errata tests or package tests but not for package-groups)

For package-groups, tests should use [FAKE_0_YUM_REPO](https://github.com/SatelliteQE/robottelo/blob/7db1851110afda569c01e8db381c69eb0c3494bf/robottelo/constants.py#L458)

I checked the `FAKE_0_YUM_REPO` and it has two package groups 1) birds 2) mammals

```console
Available Groups:
   birds
   mammals
```

**Comment by renzon** [comment]
@sghai Yesterday @rplevka gave it a look and found that this test is passing on Automation but not locally. So maybe it is some diff on config on environments.

**Comment by sghai** [comment]
@renzon : I think we are talking different issue here..

` FAKE_6_REPO` doesn't have package-groups.. so in any case that test shouldn't succeed. The issue related to `Ssh` is different.. Lets talk on IRC to get more info.

**Comment by renzon** [comment]
@sghai in fact failing test is using `FAKE_0_CUSTOM_PACKAGE_GROUP_NAME = 'birds'` and for assertion related packages: `FAKE_0_CUSTOM_PACKAGE_GROUP = ['cockateel-3.1-1.noarch', 'duck-.6-1.noarch',    'penguin-0.9.1-1.noarch',    'stork-0.12-2.noarch',]`

**Comment by renzon** [comment]
@rplevka found the issue with this. On master the test for HostCollection is using FAKE_0_YUM_REPO (https://github.com/renzon/robottelo/blob/70304fd8faee092b283853a7978312dacc1e8d2c/tests/foreman/ui/test_hostcollection.py#L412-L412) while master is using FAKE_6_YUM_REPO. It turns out the former does not have package group called "birds", so this need to be fixed when sat 6.3 become our main target again.

**Comment by renzon** [comment]
I am closing this issue in favour of #4229
